lib/runner/extensions/event.command.js

```diff
@@ -28,6 +33,60 @@ var _ = require('lodash'),
             error.name = ASSERTION_FAILURE;
         }
         return error ? serialisedError(error, true) : undefined;
+    },
+
+    /**
+     * Removes files in Request body if any.
+     *
+     * @param {Request~definition} request Request JSON representation to be sanitized
+     * @param {Function} warn function to invoke when a file is found
+     *
+     * @returns {Request~definition}
+     */
+    sanitizeFiles = function (request, warn) {
+        var sanitizedRequest,
+            sanitizedFormdata;
+
+        // do nothing if request body is empty
+        if (!request.body) {
+            return request;
+        }
+
+        // handle request body in `file` mode
+        if (request.body.mode === REQUEST_BODY_MODE_FILE) {
+            // clone the request and remove file in body
+            sanitizedRequest = _.clone(request);
+            sanitizedRequest.body.file = null;
+
+            // send a warning that file was removed
+            warn();
+
+            return sanitizedRequest;
+        }
+
+        // handle request body in `formdata` mode
+        if (request.body.mode === REQUEST_BODY_MODE_FORMDATA) {
+            // extract any file type params
```

Comment:
@kunagpal the reason I do `_.reject` first is, not all requests have file types in form data.

We should do a `clone` only when we detect a file param. Otherwise, we would be doing the `clone` even for requests without `files` which is costlier than `_.reject` of `formdata` which would be much smaller.